karoo-gp.py

```diff
@@ -435,18 +439,27 @@ def fx_karoo_pause(model):
         'tree_depth_max': model.tree_depth_max,
         'tree_depth_min': model.tree_depth_min,
         'tree_pop_max': model.tree_pop_max,
-        'gen_id': model.population.gen_id,
+        'gen_id': 0,
         'gen_max': model.gen_max,
         'tourn_size': model.tourn_size,
         'evolve_repro': model.evolve_repro,
         'evolve_point': model.evolve_point,
         'evolve_branch': model.evolve_branch,
         'evolve_cross': model.evolve_cross,
-        'fittest_dict': model.population.fittest_dict,
-        'pop_a_len': len(model.population.trees),
-        'pop_b_len': len(model.population.population_b),
+        'fittest_dict': {},
+        'pop_a_len': 0,
+        'pop_b_len':0,
         'path': model.path,
     }
+    # So it doesn't break if called before population is initialized
+    if hasattr(model, 'population'):
+        pop_dict = {
+            'gen_id': model.population.gen_id,
+            'fittest_dict': model.population.fittest_dict,
+            'pop_a_len': len(model.population.trees),
+            'pop_b_len': len(model.population.population_b),
```

Comment:
Why is the naming asymmetric?

Reply:
This seems better, but without context it's hard to tell that they are both populations.  Maybe `next_gen_pop` would be better, since it includes `pop` (and it's also shorter).